src/Microsoft.AspNetCore.Rewrite/Internal/CodeRules/RedirectRule.cs

```diff
@@ -54,7 +56,7 @@ public override void ApplyRule(RewriteContext context)
 
                 if (string.IsNullOrEmpty(newPath))
                 {
-                    response.Headers[HeaderNames.Location] = "/";
+                    response.Headers[HeaderNames.Location] = pathBase + '/';
```

Comment:
Its a problem too if it is empty, because we would be setting the location header to string.empty/